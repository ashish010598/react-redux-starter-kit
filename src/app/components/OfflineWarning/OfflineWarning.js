/**
 *
 * OfflineWarning
 *
 */

import React, { useState, useEffect, memo } from 'react';
import './OfflineWarning.scss';

export const OfflineWarning = () => {
  const [isOffline, setIsOffline] = useState(false);

  const updateOnlineStatus = () => {
    setIsOffline(!navigator.onLine);
  };

  useEffect(() => {
    window.addEventListener('online', updateOnlineStatus);
    window.addEventListener('offline', updateOnlineStatus);

    return () => {
      window.removeEventListener('online', updateOnlineStatus);
      window.removeEventListener('offline', updateOnlineStatus);
    };
  }, []);

  return isOffline ? (
    <div className="offline-warning-wrap">
      Warning! No Internet detected. The app is currently serving cached data.
    </div>
  ) : null;
};

export default memo(OfflineWarning);
