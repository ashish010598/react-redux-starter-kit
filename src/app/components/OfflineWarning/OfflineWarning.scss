.offline-warning-wrap {
  text-align: center;
  background: #fcbe63;
  color: #333;
  padding: 5px;
  font-weight: 600;
}
