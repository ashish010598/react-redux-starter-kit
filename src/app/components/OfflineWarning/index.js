import OfflineWarning from './OfflineWarning';
export default OfflineWarning;
