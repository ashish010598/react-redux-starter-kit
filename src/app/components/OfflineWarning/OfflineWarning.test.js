/**
 *
 * Tests for OfflineWarning
 *
 */

import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import OfflineWarning from '.';

describe('<OfflineWarning />', () => {
  let container = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
    act(() => {
      render(<OfflineWarning />, container);
      jest.spyOn(navigator, 'onLine', 'get').mockReturnValueOnce(false);
    });
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('renders correctly and match snapshot', async () => {
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    expect(spy).not.toHaveBeenCalled();
  });

  it('checking navigator status to false', () => {
    expect(global.navigator.onLine).toBe(false);
  });

  it('checking the visibility of no connectivity div when application is offline', () => {
    act(() => {
      const goOffline = new window.Event('offline');
      window.dispatchEvent(goOffline);
    });
    expect(
      container.firstChild.classList.contains('offline-warning-wrap'),
    ).toBe(true);
  });
});
