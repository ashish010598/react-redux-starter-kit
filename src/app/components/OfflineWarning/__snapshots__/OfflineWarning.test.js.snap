// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfflineWarning /> renders correctly and match snapshot 1`] = `""`;
